From cf4a74e799d88ffa681853e359332d5621863f1a Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sun, 13 May 2018 05:52:08 -0500
Subject: [PATCH 1/9] Fix misdetection of vlen issues when size_t is 32-bit

We pass `X_INT64_MAX - 3` into a `size_t` argument. Instead
this argument needs to be a `uint64_t`, otherwise we observe
the following failure on any platform where `size_t` is 32 bits:

ncdump -h http://geoport-dev.whoi.edu/thredds/dodsC/estofs/atlantic
ncdump: http://geoport-dev.whoi.edu/thredds/dodsC/estofs/atlantic: NetCDF: One or more variable sizes violate format constraints
---
 include/nc3internal.h | 2 +-
 libsrc/var.c          | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/nc3internal.h b/include/nc3internal.h
index 8d7af753..49ee052e 100644
--- a/include/nc3internal.h
+++ b/include/nc3internal.h
@@ -229,7 +229,7 @@ extern int
 NC_findvar(const NC_vararray *ncap, const char *name, NC_var **varpp);
 
 extern int
-NC_check_vlen(NC_var *varp, size_t vlen_max);
+NC_check_vlen(NC_var *varp, uint64_t vlen_max);
 
 extern int
 NC_lookupvar(NC3_INFO* ncp, int varid, NC_var **varp);
diff --git a/libsrc/var.c b/libsrc/var.c
index 199472d4..54ee5dab 100644
--- a/libsrc/var.c
+++ b/libsrc/var.c
@@ -515,7 +515,7 @@ out :
  * systems with LFS it should be 2^32 - 4.
  */
 int
-NC_check_vlen(NC_var *varp, size_t vlen_max) {
+NC_check_vlen(NC_var *varp, uint64_t vlen_max) {
     size_t prod=varp->xsz;	/* product of xsz and dimensions so far */
 
     int ii;
-- 
2.15.0

